validator/tools/sanitize_dashboards.js

```diff
@@ -0,0 +1,42 @@
+const { DEFINITIONS_DIR, DASHBOARD_FILE_NAME_SUFFIX, FILE_ENCODING} = require('./props');
+const {readdir} = require('fs').promises;
+const fs = require('fs');
+const path = require('path');
+
+(async () => {
+    const folderDefinitions = await readdir(DEFINITIONS_DIR, {withFileTypes: true})
+    for (let folderDefinition of folderDefinitions){
+        if(folderDefinition.isDirectory()){
+            const files = await readdir(path.resolve(DEFINITIONS_DIR, folderDefinition.name), {withFileTypes: true})
+            for(let file of  files) {
+                if(file.name.includes(DASHBOARD_FILE_NAME_SUFFIX)){
+                    let filePath = path.resolve(DEFINITIONS_DIR, folderDefinition.name, file.name)
+                    fs.readFile(filePath, FILE_ENCODING, function(err, fileContent){
+                        if (err){
+                            console.error(`Error reading ${file.name} in ${folderDefinition.name}`)
+                            process.exit(1)
+                        }
+
+                        let newFileContent = fileContent
+                            .replace(/\"accountId\"\s*:\s*\d+\s*/g , '"accountId": 0') // Anonymize account ID
+                            .replace(/^.+\"linkedEntityGuids\".+\n?/mg , '') // Remove linkedEntityGuids
+                            .replace(/^.+\"permissions\".+\n?/mg , '') // Remove permissions
+                            .replace(/\,(?!\s*?[\{\[\"\'\w])/g, ''); // Remove trailing commas
```

Comment:
You must be getting handy at the auld regex at this stage :D

Reply:
😆 😭